backend/app/models/types.py

```diff
@@ -59,4 +60,16 @@ def __eq__(self, other: Any) -> bool:
         return isinstance(other, EncryptedStr) and self.get_decrypted_value() == other.get_decrypted_value()
 
     def get_decrypted_value(self) -> str:
-        return security.decrypt_password(self)
+        decrypted_value: str = security.decrypt_password(self)
+        # if the decrypted_value matches the regex of a secrets provider,
+        # e.g. AWS Secrets Manager, GCP Secret Manager, Azure Key Vault
+        # get secret value from provider
+        try:
```

Comment:
Could be interesting to put this whole logic in a dedicated module/function so we don't pollute too much the model.

Reply:
Agreed, this could be handy if we decide to support retrieving secrets for non EncryptedStr types.